src/main/java/pl/javastart/task/Room.java

```diff
@@ -0,0 +1,63 @@
+package pl.javastart.task;
+
+public class Room {
+    private double area;
+    private double currentTemperature;
+    private double minimumTemperature;
+    boolean airConditioning;
+
+    public Room(double area, double currentTemperature) {
+        this.area = area;
+        this.currentTemperature = currentTemperature;
+    }
+
+    public Room(double area, double currentTemperature, boolean airConditioning) {
+        this(area, currentTemperature);
+        this.airConditioning = airConditioning;
+        if (airConditioning) {
+            this.minimumTemperature = 16.0;
+        }
+    }
+
+    public Room(double area, double currentTemperature, double minimumTemperature, boolean airConditioning) {
+        this(area, currentTemperature);
+        this.airConditioning = airConditioning;
+        if (airConditioning) {
+            this.minimumTemperature = minimumTemperature;
+        }
+    }
+
+    public double getArea() {
+        return area;
+    }
+
+    public double getCurrentTemperature() {
+        return currentTemperature;
+    }
+
+    public double getMinimumTemperature() {
+        return minimumTemperature;
+    }
+
+    public boolean isAirConditioning() {
+        return airConditioning;
+    }
+
+    public void setMinimumTemperature(double minimumTemperature) {
+        this.minimumTemperature = minimumTemperature;
+    }
+
+    public boolean lowerTemperature() {
+        boolean lowerTemperatureSuccessful = false;
+        if (!airConditioning) {
+            return lowerTemperatureSuccessful;
+        }
+        if (currentTemperature - 1.0 >= minimumTemperature) {
+            currentTemperature -= 1.0;
+            lowerTemperatureSuccessful = true;
+        } else {
+            currentTemperature = minimumTemperature;
+        }

```

Comment:
Jeżeli temperatura aktualna będzie np. 16.5, a minimalna to 16, to mamy tutaj też obniżenie temperatury i zgodnie z treścią zadania też powinieneś zwrócić w takiej sytuacji true

Reply:
"Metoda powinna zwracać true, jeżeli udało się obniżyć temperaturę, lub false jeżeli w pokoju nie ma klimatyzatora lub osiągnięto temperaturę graniczną". Zrozumiałem to w ten sposób, że true gdy udało się obniżyć o 1 stopień, a false także gdy udało się obniżyć, ale mniej niż o 1 stopień, ponieważ o 1 stopień nie udało się obniżyć, bo wcześniej osiągnięto temperaturę graniczną. Poprawiam anyway.

Reply:
Jest różnica między:
Aktualna temperatura 18, minimalna 16 -> obniżamy temperaturę o 1 stopień -> true
Aktualna temperatura 16.5, minimalna 16 -> obniżamy temperaturę o 0,5 stopnia -> true
Aktualna temperatura 16, minimalna 16 -> osiągnięto temperaturę graniczną, nie obniżamy jej dalej -> false

Reply:
Jasne. Poprawione wczoraj